primer3/wrappers.py

```diff
@@ -244,86 +297,63 @@ def assessOligo(seq):
+    data_dict = OrderedDict()
+    for line in boulder_str.decode('utf-8').split('\n'):
+        try:
+            k, v = line.strip().split('=')
+            data_dict[k] = v
+        except ValueError:
+            pass
+    return data_dict
+
+
+def parseMultiRecordBoulderIO(boulder_str: str) -> List[OrderedDict[str, str]]:
+    data_dicts = []
+    for record in re.split('=\r?\n', boulder_str):
+        if record == '':
+            continue
         data_dict = OrderedDict()
-        for line in boulder_str.decode('utf-8').split('\n'):
+        for line in record.split('\n'):
             try:
                 k, v = line.strip().split('=')
                 data_dict[k] = v
             except ValueError:
                 pass
-        return data_dict
-
-    def _parseMultiRecordBoulderIO(boulder_str):
-        data_dicts = []
-        for record in re.split('=\r?\n', boulder_str):
-            if record == '':
-                continue
-            data_dict = OrderedDict()
-            for line in record.split('\n'):
-                try:
-                    k, v = line.strip().split('=')
-                    data_dict[k] = v
-                except ValueError:
-                    pass
-            data_dicts.append(data_dict)
-        return data_dicts
-
-else:
-
-    def _formatBoulderIO(p3_args):
-        boulder_str = ''.join([
-            '{}={}\n'.format(k, v) for k, v in
-            p3_args.items()
-        ])
-        boulder_str += '=\n'
-        return boulder_str
-
-    def _parseBoulderIO(boulder_str):
-        data_dict = OrderedDict()
-        for line in boulder_str.split('\n'):
-            try:
-                k, v = line.strip().split('=')
-                data_dict[k] = v
-            except ValueError:
-                pass
-        return data_dict
-
-    def _parseMultiRecordBoulderIO(boulder_str):
-        data_dicts = []
-        for record in re.split('=\r?\n', boulder_str):
-            if record == '':
-                continue
-            data_dict = OrderedDict()
-            for line in record.split('\n'):
-                try:
-                    k, v = line.strip().split('=')
-                    data_dict[k] = v
-                except ValueError:
-                    pass
-            data_dicts.append(data_dict)
-        return data_dicts
-
-
-def designPrimers(p3_args, input_log=None, output_log=None, err_log=None):
+        data_dicts.append(data_dict)
+    return data_dicts
+
+
+def designPrimers(
+        p3_args: Dict[str, Any],
+        input_log: Optional[io.BufferedWriter] = None,
+        output_log: Optional[io.BufferedWriter] = None,
+        err_log: Optional[io.BufferedWriter] = None,
+) -> Dict[str, Any]:
     ''' Return the raw primer3_core output for the provided primer3 args.
 
-    Returns an ordered dict of the boulderIO-format primer3 output file
+    Args:
+        p3_args: Dictionarty of arguments for the primer3
+        input_log: Optional log input file descriptor
```

Comment:
Open file handle? Looks like it could be an `io.StringIO` object (or analogous as well)?

Reply:
`_formatBoulderIO` always returns bytes so it can't be an `io.StringIO` I think?

Reply:
Ah good point, I guess it could be an `io.BytesIO` then. Either way pretty insignificant.